projects/plugins/jetpack/modules/subscriptions/subscribe-overlay/class-jetpack-subscribe-overlay.php

```diff
@@ -0,0 +1,208 @@
+<?php
+/**
+ * Adds support for Jetpack Subscribe Overlay feature
+ *
+ * @package automattic/jetpack-subscriptions
+ * @since $$next-version$$
+ */
+
+/**
+ * Jetpack_Subscribe_Overlay class.
+ */
+class Jetpack_Subscribe_Overlay {
+	/**
+	 * Jetpack_Subscribe_Overlay singleton instance.
+	 *
+	 * @var Jetpack_Subscribe_Overlay|null
+	 */
+	private static $instance;
+
+	/**
+	 * Jetpack_Subscribe_Overlay instance init.
+	 */
+	public static function init() {
+		if ( self::$instance === null ) {
+			self::$instance = new Jetpack_Subscribe_Overlay();
+		}
+
+		return self::$instance;
+	}
+
+	const BLOCK_TEMPLATE_PART_SLUG = 'jetpack-subscribe-overlay';
+
+	/**
+	 * Jetpack_Subscribe_Overlay class constructor.
+	 */
+	public function __construct() {
+		if ( apply_filters( 'jetpack_subscribe_overlay_enabled', false ) ) {
+			add_action( 'wp_enqueue_scripts', array( $this, 'enqueue_assets' ) );
+			add_action( 'wp_footer', array( $this, 'add_subscribe_overlay_to_frontend' ) );
+		}
+
+		add_filter( 'get_block_template', array( $this, 'get_block_template_filter' ), 10, 3 );
```

Comment:
shouldn't this be under the feature flag too?

Reply:
I think it's safe to leave it as it is. It allows the user to edit an overlay template if he has a link but there is no link to it anywhere.